cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -42,8 +45,49 @@ import (
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 )
 
+func getK8sVersionFromUserInput(flags *applyPlanFlags, args []string, versionIsMandatory bool) (string, error) {
+	var userVersion string
+
+	// If the version is specified in config file, pick up that value.
+	if flags.cfgPath != "" {
+		// Note that cfg isn't preserved here, it's just an one-off to populate userVersion based on --config
+		cfg, err := configutil.LoadInitConfigurationFromFile(flags.cfgPath)
+		if err != nil {
+			return "", err
+		}
+
+		if cfg.KubernetesVersion != "" {
```

Comment:
The if statement does not make any difference if `cfg.KubernetesVersion == ""`, because `userVersion` should be empty at that moment. Just leave `userVersion = cfg.KubernetesVersion`.

Reply:
Thanks! Fixed.